File: Easy/509-fibonacci-number/try-6.cpp
/*
 * Problem #509: Fibonacci Number
 * Difficulty: Easy
 * Submission: Try 6
 * status: Accepted
 * Language: cpp
 * Date: 10/01/2025, 21:25:56
 * Link: https://leetcode.com/problems/fibonacci-number/
 */

class Solution {
public:
    int fib(int n) {
        vector<int> v(n+1,0);
        if (n==0){
            return 0;
        }
        if (n==1){
            return 1;
        }
        v[0] = 0;
        v[1] = 1;
        for(int i = 2;i<n+1;i++){
            v[i] = v[i-1]+v[i-2];
        }
        return v[n];

    }
};

File: Easy/1-two-sum/try-6.py
/*
 * Problem #1: Two Sum
 * Difficulty: Easy
 * Submission: Try 6
 * status: Accepted
 * Language: python3
 * Date: 23/11/2024, 13:38:44
 * Link: https://leetcode.com/problems/two-sum/
 */

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            comp=target-nums[i]
            if comp in map:
                return [map[comp],i]
            map[nums[i]]=i
        return []
        
